### PR TITLE
1324: create mapping records when users are created

### PR DESCRIPTION
**Pull request by rachaelparris**
Plus an unrelated test for coverage.

Logic in the persistence layer doesn't feel right, but I'm not sure how to do this differently. Open to other suggestions.

### CODE REVIEW

File: shared/src/persistence/dynamo/users/createUser.js [hunk 1]
```diff
@@ -1,7 +1,59 @@
...
+      ...user,
+      userId,
+    },
+    applicationContext,
+  });
+
+  if (
+    (user.role === 'practitioner' || user.role === 'respondent') &&
+    user.name &&
+    user.barNumber !== '0'
```
Comment (matthopson) [inline shared/src/persistence/dynamo/users/createUser.js:34]: Is this a default value or something?

Comment (rachaelparris) [reply shared/src/persistence/dynamo/users/createUser.js:34]: Yeah, because Cognito wouldn't allow it to be empty. 🤷‍♀

Comment (matthopson) [reply shared/src/persistence/dynamo/users/createUser.js:34]: That's bunk.

**Review by matthopson — APPROVED** [review]
I wonder if there'd be any value in adding some of the role / bar number logic on the User entity, so you can just hit a helper. Additionally this could strip / reformat that bar number based on its value or role when created. Just a thought.

File: shared/src/persistence/dynamo/users/createUser.js (hunk 1 above)
Comment (codyseibert) [reply shared/src/persistence/dynamo/users/createUser.js:34]: I don't think the barNumber should be stored on the cognito user object, only on dynamo.

Comment (rachaelparris) [reply shared/src/persistence/dynamo/users/createUser.js:34]: Sorry, Dynamo wouldn't allow it to be empty. Not Cognito.

**Comment by rachaelparris** [comment]
> I wonder if there'd be any value in adding some of the role / bar number logic on the User entity, so you can just hit a helper. Additionally this could strip / reformat that bar number based on its value or role when created. Just a thought.

I'll look into refactoring this. Thanks for the feedback!

**Pull request merged**